src/utils/progress_utils.py

```
@@ -0,0 +1,207 @@
+        
+        Args:
+            ctx: FastMCP context for progress reporting
+            tool_name: Name of the tool for logging
+            total_steps: Total progress steps (default: 100)
+        """
+        self.ctx = ctx
+        self.tool_name = tool_name
+        self.total_steps = total_steps
+        self.current_progress = 0
+        
+    async def report(self, progress: int, message: str = None, force: bool = False):
```

Ruff (0.12.2)

Missing return type annotation for public function `report`

Add return type annotation: `None`

(ANN201)